python-avd/pyavd/_eos_designs/structured_config/network_services/utils_wan.py

```diff
@@ -37,184 +37,325 @@ def _filtered_wan_vrfs(self: AvdStructuredConfigNetworkServicesProtocol) -> EosD
+
+        if not output_policy.matches:
             # The policy is empty but should be assigned to a VRF
             msg = (
-                f"The policy `wan_virtual_topologies.policies[{policy['name']}]` cannot match any traffic but is assigned to a VRF. "
+                f"The policy `wan_virtual_topologies.policies[{policy.name}]` cannot match any traffic but is assigned to a VRF. "
                 "Make sure at least one path-group present on the device is used in the policy."
             )
             raise AristaAvdError(msg)
 
-        policy["matches"] = matches
-        policy["default_match"] = default_match
+        self.structured_config.router_adaptive_virtual_topology.policies.append(output_policy)
+        self.structured_config.router_adaptive_virtual_topology.vrfs.append(output_vrf)
+
+    def _verify_policy_default_match(self: AvdStructuredConfigNetworkServicesProtocol, policy: EosDesigns.WanVirtualTopologies.PoliciesItem) -> None:
+        """
+        Verifies the policy has a proper default_match definition.
+
+        Checks that:
+            * the default_virtual_topology is defined.
+            * either drop_unmatched must be set or some path_groups must be defined.
+
+        Raises:
+            AristaAvdInvalidInputsError: if any criteria is not met.
+        """
+        if not policy.default_virtual_topology:
+            msg = f"wan_virtual_topologies.policies[{policy.name}].default_virtual_toplogy."
+            raise AristaAvdInvalidInputsError(msg)
+
+        if not policy.default_virtual_topology.drop_unmatched:
+            context_path = f"wan_virtual_topologies.policies[{policy.name}].default_virtual_topology"
+            # Verify that path_groups are set or raise
+            # TODO: this functionality could be added to schema
+            if not policy.default_virtual_topology.path_groups:
+                msg = f"Either 'drop_unmatched' or 'path_groups' must be set under '{context_path}'."
+                raise AristaAvdInvalidInputsError(msg)
+
+    def _verify_internet_exit_policy(self: AvdStructuredConfigNetworkServicesProtocol, ie_policy_name: str, policy_name: str) -> None:
+        """Check if the Internet Exit policy name is configured.
+
+        If not raise an AristaAvdInvalidInputsError.
+
+        TODO: This used to check that there is an interface to make it valid.
+        """
+        if ie_policy_name not in self.inputs.cv_pathfinder_internet_exit_policies:
+            msg = (
+                f"The internet exit policy {ie_policy_name} configured under "
+                f"`wan_virtual_topologies.policies[name={policy_name}].internet_exit.policy` "
+                "is not defined under `cv_pathfinder_internet_exit_policies`."
+            )
+            raise AristaAvdInvalidInputsError(msg)
+
+    local_internet_exit_connections: dict[str, bool] | None = None
+    """Poor-mans cache to only check local interfaces for an internet exit policy once."""
```

Comment:
We should write our own cheap decorator for this pattern. (Outside of this work ofc.)